### PR TITLE
CONN-1952 remove Sonarqube NullPointerException issues

### PR DESCRIPTION
**Pull request by ptambellini01**

### CODE REVIEW

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/asyncmsgs/dao/AsyncMsgRecordDao.java [hunk 1]
```diff
@@ -656,11 +656,11 @@ private Date calculateExpirationValue(long value, String units) {
         // Convert the long to a Long Object and change the sign to negative so our query value ends up in the past.
         Long longObj = -value;
 
-        if (units.equalsIgnoreCase(NhincConstants.ASYNC_DB_REC_EXP_VAL_UNITS_SEC)) {
+        if (units.equalsIgnoreCase(NhincConstants.ASYNC_DB_REC_EXP_VAL_UNITS_SEC) && units != null) {
```
Comment (mhpnguyen) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/asyncmsgs/dao/AsyncMsgRecordDao.java]: use this instead NhincConstants.ASYNC_DB_REC_EXP_VAL_UNITS_SEC.equalsIgnoreCase(units) and remove units!= null

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/asyncmsgs/dao/AsyncMsgRecordDao.java [hunk 2]
```diff
@@ -656,11 +656,11 @@ private Date calculateExpirationValue(long value, String units) {
...
             currentTime.add(Calendar.SECOND, longObj.intValue());
-        } else if (units.equalsIgnoreCase(NhincConstants.ASYNC_DB_REC_EXP_VAL_UNITS_MIN)) {
+        } else if (units.equalsIgnoreCase(NhincConstants.ASYNC_DB_REC_EXP_VAL_UNITS_MIN) && units != null) {
```
Comment (mhpnguyen) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/asyncmsgs/dao/AsyncMsgRecordDao.java]: similar like above comments

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/asyncmsgs/dao/AsyncMsgRecordDao.java [hunk 3]
```diff
@@ -656,11 +656,11 @@ private Date calculateExpirationValue(long value, String units) {
...
             currentTime.add(Calendar.MINUTE, longObj.intValue());
-        } else if (units.equalsIgnoreCase(NhincConstants.ASYNC_DB_REC_EXP_VAL_UNITS_HOUR)) {
+        } else if (units.equalsIgnoreCase(NhincConstants.ASYNC_DB_REC_EXP_VAL_UNITS_HOUR) && units != null) {
```
Comment (mhpnguyen) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/asyncmsgs/dao/AsyncMsgRecordDao.java]: check above comments

**Review by mhpnguyen — APPROVED** [review]

**Review by mhpnguyen — CHANGES_REQUESTED** [review]
Please see my comments

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/opensaml/extraction/OpenSAMLAssertionExtractorImpl.java [hunk 4]
```diff
@@ -105,7 +104,9 @@ public final AssertionType extractSAMLAssertion(final Element element) {
 
         AssertionType target = initializeAssertion();
         // Populate Issuer
-        populateIssuer(saml2Assertion, target);
+        if (saml2Assertion != null) {
+            populateIssuer(saml2Assertion, target);
+        }
         // Populate the Subject Information
         populateSubject(saml2Assertion, target);
```
Comment (TabassumJafri) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/opensaml/extraction/OpenSAMLAssertionExtractorImpl.java:111]: populateSubject .. shouldn't this cause a null dereference if saml2Assertion is null?

**Review by TabassumJafri — CHANGES_REQUESTED** [review]

**Review by mhpnguyen — APPROVED** [review]

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/opensaml/extraction/OpenSAMLAssertionExtractorImpl.java (hunk 4 above)
Comment (ptambellini01) [reply Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/opensaml/extraction/OpenSAMLAssertionExtractorImpl.java:111]: populate Subject can return a null and will not cause a null dereference.

**Review by TabassumJafri — APPROVED** [review]

**Pull request merged**